File: lld_design_primer_part2/rate_limiter/Approach_2/Logger.java
package Approach_2;

import java.util.HashMap;

class Logger {
    HashMap<String, Integer> nextEligibleTime;

    /** Initialize your data structure here. */
    public Logger() {
        nextEligibleTime = new HashMap<>();
    }

    public boolean shouldPrintMessage(int timestamp, String message) {
        if (timestamp < nextEligibleTime.getOrDefault(message, 0)) {
            return false;
        }
        nextEligibleTime.put(message, timestamp + 10);
        return true;
    }
}

File: lld_design_primer_part2/rate_limiter/Approach_2/Logger.py
class Logger:
    def __init__(self):
        self.nextEligibleTime = {}

    def shouldPrintMessage(self, timestamp, message):
        if timestamp < self.nextEligibleTime.get(message, 0):
            return False
        self.nextEligibleTime[message] = timestamp + 10
        return True
